src/Equations/damaged-elastic/Kernels/Local.cpp

```diff
@@ -515,3 +501,565 @@ unsigned seissol::kernels::Local::bytesIntegral() {
 
   return reals * sizeof(real);
 }
+
+void seissol::kernels::Local::computeNonLinearRusanovFlux(
+    const CellMaterialData* materialData,
+    unsigned int l_cell,
+    unsigned int side,
+    const double* timeWeights,
+    const real* qIPlus,
+    const real* qIMinus,
+    real* rusanovFluxP,
+    const LocalIntegrationData* localIntegration) {
+  using namespace seissol::dr::misc::quantity_indices;
+  const unsigned DAM = 9;
+  const unsigned BRE = 10;
+
+  const real lambda0P = materialData[l_cell].local.lambda0;
+  const real mu0P = materialData[l_cell].local.mu0;
+  const real rho0P = materialData[l_cell].local.rho;
+
+  const real lambda0M = materialData[l_cell].neighbor[side].lambda0;
+  const real mu0M = materialData[l_cell].neighbor[side].mu0;
+  const real rho0M = materialData[l_cell].neighbor[side].rho;
+
+  const real epsInitxx = m_damagedElasticParameters->epsInitxx;
+  const real epsInityy = m_damagedElasticParameters->epsInityy;
+  const real epsInitzz = m_damagedElasticParameters->epsInitzz;
+  const real epsInitxy = m_damagedElasticParameters->epsInitxy;
+  const real epsInityz = m_damagedElasticParameters->epsInityz;
+  const real epsInitzx = m_damagedElasticParameters->epsInitzx;
+
+  const real aB0 = m_damagedElasticParameters->aB0;
+  const real aB1 = m_damagedElasticParameters->aB1;
+  const real aB2 = m_damagedElasticParameters->aB2;
+  const real aB3 = m_damagedElasticParameters->aB3;
+
+  real lambdaMax = 1.0 * std::sqrt((lambda0P + 2 * mu0P) / rho0P);
+  real sxxP, syyP, szzP, sxyP, syzP, szxP, sxxM, syyM, szzM, sxyM, syzM, szxM;
+  seissol::kernels::Time m_timeKernel;
+  m_timeKernel.setDamagedElasticParameters(m_damagedElasticParameters);
+
+  for (unsigned o = 0; o < CONVERGENCE_ORDER; ++o) {
+    auto weight = timeWeights[o];
+
+    for (unsigned i = 0; i < seissol::dr::misc::numPaddedPoints; i++) {
+
+      real EspIp, EspIIp, xip, EspIm, EspIIm, xim;
+
+      std::tie(EspIp, EspIIp, xip) = m_timeKernel.calculateEsp(
+          &qIPlus[o * seissol::dr::misc::numQuantities * seissol::dr::misc::numPaddedPoints +
+                  XX * seissol::dr::misc::numPaddedPoints],
+          &qIPlus[o * seissol::dr::misc::numQuantities * seissol::dr::misc::numPaddedPoints +
+                  YY * seissol::dr::misc::numPaddedPoints],
+          &qIPlus[o * seissol::dr::misc::numQuantities * seissol::dr::misc::numPaddedPoints +
+                  ZZ * seissol::dr::misc::numPaddedPoints],
+          &qIPlus[o * seissol::dr::misc::numQuantities * seissol::dr::misc::numPaddedPoints +
+                  XY * seissol::dr::misc::numPaddedPoints],
+          &qIPlus[o * seissol::dr::misc::numQuantities * seissol::dr::misc::numPaddedPoints +
+                  YZ * seissol::dr::misc::numPaddedPoints],
+          &qIPlus[o * seissol::dr::misc::numQuantities * seissol::dr::misc::numPaddedPoints +
+                  XZ * seissol::dr::misc::numPaddedPoints],
```

Comment:
You could just pass `qIPlus` and do the pointer arithmetic in the function?

Reply:
The function `calculateEsp` is used in other places where the pointer is different along with the shape used. I don't think this is possible to be easily done without overloading the function definitions as per wherever it is used. `Time.cpp`, lines 210-230, for example

Reply:
The lambda function strategy seems reasonable and is working well.

Reply:
Used the lambda functions